### PR TITLE
add update-lifted and verify-lifted

### LINKED ISSUE #1652: Automatic update lifted/doc.go
**What would you like to be added**:
`lifted/doc.go` seems to be updated manually. It is better to do it automatically.

We can add lifted info in comments, like:
```
//+source: https://github.com/kubernetes/kubernetes/blob/release-1.23/pkg/apis/core/helper/helpers.go#L57-L61
//+changed: no

// IsQuotaHugePageResourceName returns true if the resource name has the quota
// related huge page resource prefix.
func IsQuotaHugePageResourceName(name core.ResourceName) bool {
	return strings.HasPrefix(string(name), core.ResourceHugePagesPrefix) || strings.HasPrefix(string(name), core.ResourceRequestsHugePagesPrefix)
}
```

Then we can parse it with `AST` and generate `doc.go`. 
`update-lifted.sh` and `verify-lifted.sh` can be added in hack.

**Why is this needed**:
Avoid `lifted/doc.go` out of date.

### PR DESCRIPTION
**Pull request by ikaven1024**
Signed-off-by: yingjinhui <yingjinhui@didiglobal.com>

**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
Update and verify `lifted/doc.go` automatically.

**Which issue(s) this PR fixes**:
Fixes #1652

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

### CODE REVIEW

**Comment by ikaven1024** [comment]
@XiShanYongYe-Chang @mrlihanbo @lonelyCZ

File: hack/verify-lifted.sh [hunk 1]
```diff
@@ -0,0 +1,20 @@
+#!/usr/bin/env bash
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+SCRIPT_ROOT=$(dirname "${BASH_SOURCE[0]}")/..
+
+doc_file="pkg/util/lifted/doc.go"
+"$SCRIPT_ROOT/hack/update-lifted.sh"
```
Comment (XiShanYongYe-Chang) [inline hack/verify-lifted.sh:10]: Is it possible to create a temporary file like any other verify script and then restore it after the comparison?

**Review by XiShanYongYe-Chang — COMMENTED** [review]
This looks great, thanks for your contribution.

File: hack/verify-lifted.sh (hunk 1 above)
Comment (ikaven1024) [reply hack/verify-lifted.sh:10]: ok~

**Comment by lonelyCZ** [comment]
LGTM, this is great. I think whether we could add a usage example or document for this feature later, which can help contributors to learn about this specification.

**Comment by ikaven1024** [comment]
> I think whether we could add a usage example or document for this feature later,

add it at another pr.

**Comment by ikaven1024** [comment]
@XiShanYongYe-Chang Plz take a look again.

**Review by XiShanYongYe-Chang — COMMENTED** [review]
/lgtm

**Comment by ikaven1024** [comment]
@RainbowMango could you approve it?

**Comment by ikaven1024** [comment]
@RainbowMango  PTAL

**Comment by RainbowMango** [comment]
OK. Try my best to look at it today. Sorry for the delay.

File: hack/tools/lifted/lifted.go [hunk 2]
```diff
@@ -0,0 +1,243 @@
+package main
```
Comment (RainbowMango) [inline hack/tools/lifted/lifted.go:1]: What do you think about the name `lifted-gen`? Just like `client-gen`/`deepcopy-gen`? :)

`hack/tools/lifted-gen/main.go`.

File: hack/tools/lifted/lifted.go [hunk 3]
```diff
@@ -0,0 +1,243 @@
...
+	"path/filepath"
+	"sort"
+	"strings"
+
+	"github.com/olekukonko/tablewriter"
+)
+
+const (
+	liftedCommentPrefix  = "//+lifted:"
+	liftedCommentChanged = "changed"
```
[the remark this answers is not in the document]
Comment (RainbowMango) [reply hack/tools/lifted/lifted.go:20]: No better ideas. I'd prefer the second one between the two.

**Comment by ikaven1024** [comment]
@RainbowMango updated, and PTAL.

**Comment by ikaven1024** [comment]
shall we moving this forward now 😂 @RainbowMango?

**Comment by RainbowMango** [comment]
Yeah. Will look at it today.(try my best :))

**Comment by RainbowMango** [comment]
We are ready to go now. Just a question, should we enable the verification at workflow?
https://github.com/karmada-io/karmada/blob/2a9d10628819a47a6da6ea4f637c24a1c6847066/.github/workflows/ci.yml#L24-L50

**Comment by ikaven1024** [comment]
> should we enable the verification at workflow?

lgtm.    
I thought `verify-all.sh` is called in ci.

**Review by RainbowMango — APPROVED** [review]
/lgtm
/approve

**Pull request merged**